### DISCUSSION TITLE
Documentation needs to have ability to switch between stable, past versions, and upcoming or experimental versions

### DISCUSSION THREAD

**Discussion by ricketyboo** [nrwl/nx#23345] [Feature Requests]
### Documentation issue

- [ ] Reporting a typo
- [ ] Reporting a documentation bug
- [x] Documentation improvement
- [x] Documentation feedback

### Is there a specific documentation page you are reporting?

It's maybe more than  just one page, but https://nx.dev/nx-api/vue is an example causing confusion that led to https://github.com/nrwl/nx/issues/21161

### Additional context or description

The documentation seems to be written and published based on work that isn't actually stable or in latest yet. Linked is a vue document that suggests a vue preset exists, but when reporting a bug with the preset, I was told "use the canary version". I saw another issue where someone was querying  use of Nuxt from what looked like the nuxt plugin documentation and was told something along the lines of "that is coming soon, we haven't released it yet!". (Though to be fair the nuxt documentation doesn't imply as strongly a readiness as nx/vue did)

I don't know how widespread this is, or if this is just a mistake and something was published too soon, but I think it would be beneficial to make sure the documentation is tied to the versions we're on (and indeed, have a version toggle on the documentation so it's possible for people who aren't quite able to upgrade to latest can see their version) and include any non stable "pending release" stuff in a "canary" or "release candidate" or whatever we call it branch.

It's a tough sell to my organisation to get any new technology on board and one of the battles I have to do when trying to sell things is around documentation and how easy it is to do a proof of concept. I already know I want to use NX (I used it for years on angular, in a previous life) and am happy to push past the stuff that doesn't work out of the box, but I suspect that many people in similar organisational processes that may be evaluating for the first time will just see the docs, run the command, see it break and go "I guess this isn't a very stable product, you know what open source is like" and bounce.

Examples of other open source projects showing what I mean, Angular being the best example.

Angular
![image](https://github.com/nrwl/nx/assets/4494839/9c7b1152-d266-4885-8e61-252739a1bde0)

Tailwind
![image](https://github.com/nrwl/nx/assets/4494839/305f44b8-718b-44d2-9bb3-cdd5d92161db)

https://getbootstrap.com/docs/versions/

etc

**Comment by FrozenPandaz** [comment]
I've converted this to a discussion so we can discuss more easily.

cc @isaacplmann
